cmd/syncComparator.go

```diff
@@ -109,11 +152,47 @@ func (f *syncSourceComparator) processIfNecessary(sourceObject StoredObject) err
 	if present {
 		defer delete(f.destinationIndex.indexMap, relPath)
 
-		// if destination is stale, schedule source for transfer
-		if f.disableComparison || sourceObject.isMoreRecentThan(destinationObjectInMap, f.preferSMBTime) {
+    // if destination is stale, schedule source for transfer
+		if f.disableComparison {
 			return f.copyTransferScheduler(sourceObject)
 		}
-		// skip if source is more recent
+
+		if f.comparisonHashType != common.ESyncHashType.None() && sourceObject.entityType == common.EEntityType.File() {
+			switch f.comparisonHashType {
+			case common.ESyncHashType.MD5():
+				if sourceObject.md5 == nil {
+					if azcopyScanningLogger != nil {
+						azcopyScanningLogger.Log(pipeline.LogWarning, fmt.Sprintf("File %s skipped, because the source does not have a hash.", sourceObject.relativePath))
```

Comment:
same as above comment, too much duplication, why don't we move this out in a separate method